packages/manager/apps/ips/src/data/hooks/ip/useIpHasServicesNotAttached.ts

```diff
@@ -0,0 +1,33 @@
+import { useGetProductService } from '../useGetProductService';
+
+// List of all relevant product paths and categories to check except { path: '/dedicated/housing', category: 'HOUSING' },
+const PRODUCT_PATHS_AND_CATEGORIES = [
+  { path: '/cloud/project', category: 'CLOUD' },
+  { path: '/dedicated/server', category: 'DEDICATED' },
+  { path: '/vps', category: 'VPS' },
+  { path: '/vrack', category: 'VRACK' },
+  { path: '/dedicatedCloud', category: 'PRIVATE_CLOUD' },
+  { path: '/ipLoadbalancing', category: 'IP_LOAD_BALANCING' },
+];
+
+export type UseIpHasServicesNotAttachedParams = {
+  serviceName?: string;
+};
+
+export const useIpHasServicesNotAttached = ({
+  serviceName,
+}: UseIpHasServicesNotAttachedParams) => {
+  // Fetch all product services for all paths/categories
+  const productServicesList = PRODUCT_PATHS_AND_CATEGORIES.map((entry) =>
+    useGetProductService(entry),
+  );
```

Comment:
We should have a useQueries for multiple queries, not use a hook inside a `.map` I think

Reply:
Even though the array is static, I don't think it's a good idea

Reply:
Maybe we should have another hook `useGetProductService**s**` that accept an array and returns the result of useQueries